discovery_engine_core/core/src/storage/sqlite/utils.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use sqlx::{Database, Encode, QueryBuilder, Type};
+
+pub(super) trait SqlxPushTupleExt<'args, DB: Database> {
```

Comment:
❓ since this is a `sqlx` utitlity and not a `sqlite` utility, should we put this module somewhere else? maybe as `crate::storage::sqlx` or sth similar?

Reply:
I considered it, but thought that given that we only have sqlite for not it is for now easier to place it into the sqlite `utils` module and move it when needed.

But thinking about it more it probably is better to move it now to avoid updating imports.